api/src/task/notifications/opportunity_notifcation.py

```diff
@@ -1,86 +1,443 @@
+                    stmt = ELIGIBILITY_FIELDS["applicant_types"]
+                    if added:
+                        eligibility_section += (
+                            f"{BULLET_POINTS_STYLING} Additional {stmt} {list(added)}.<br>"
+                        )
+                    if removed:
+                        eligibility_section += (
+                            f"{BULLET_POINTS_STYLING} Removed {stmt} {list(removed)}.<br>"
+                        )
+
+                if field == "applicant_eligibility_description":
+                    stmt = f"{BULLET_POINTS_STYLING} {ELIGIBILITY_FIELDS["applicant_eligibility_description"]}"
+
+                    if not before and after:
+                        eligibility_section += f"{stmt} added.<br>"
+                    elif before and not after:
+                        eligibility_section += f"{stmt} removed.<br>"
+                    else:
+                        eligibility_section += f"{stmt} updated.<br>"
+
+            if field in GRANTOR_CONTACT_INFORMATION_FIELDS:
+                if not grantor_section:
+                    grantor_section =  SECTION_STYLING.format("Grantor contact information")
+                grantor_section += f"{BULLET_POINTS_STYLING} {GRANTOR_CONTACT_INFORMATION_FIELDS[field]} {after}.<br>"
+
+            if field in DOCUMENTS_FIELDS:
+                if not documents_section:
+                    documents_section = SECTION_STYLING.format("Documents")
+                if field == "attachments":
+
+                    before_set = set(json.dumps(d, sort_keys=True) for d in before)
+                    after_set = set(json.dumps(d, sort_keys=True) for d in after)
+
+                    if after_set - before_set:
+                        documents_section += f"{BULLET_POINTS_STYLING} {DOCUMENTS_FIELDS["attachments"]} added.<br>"
+                    if before_set - after_set:
+                        documents_section += f"{BULLET_POINTS_STYLING} {DOCUMENTS_FIELDS["attachments"]} removed.<br>"
+
+                if field == "additional_info_url":
+                    stmt = f"{BULLET_POINTS_STYLING} {DOCUMENTS_FIELDS["additional_info_url"]}"
+                    if not before and after:
+                        eligibility_section += f"{stmt} added.<br>"
+                    elif before and not after:
+                        eligibility_section += f"{stmt} removed.<br>"
+
+        change_sections = (
+                status_section
+                + important_section
+                + award_section
+                + category_section
+                + description_section
+                + eligibility_section
+                + grantor_section
+                + documents_section
+        )
+
+        return   change_sections
+
+    def _build_notification_content(
+        self, updated_saved_opps: dict[UUID, dict[str, OpportunityVersion]]
+    ) -> tuple[str ,str] | tuple[str,str, list] | tuple[None,None, None]:
```

Comment:
If a function is returning that many tuples of generic types, I suggest making a class that contains those so it's obvious what we're working with